api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -61,3 +64,18 @@ type KubeadmConfigList struct {
 func init() {
 	SchemeBuilder.Register(&KubeadmConfig{}, &KubeadmConfigList{})
 }
+
+// Files defines the input for generating write_files in cloud-init.
+type Files struct {
+	// Path specifies the full path on disk where to store the file.
+	Path string `json:"path"`
+
+	// Owner specifies the ownership of the file, e.g. "root:root".
+	Owner string `json:"owner"`
+
+	// Permissions specifies the permissions to assign to the file, e.g. "0640".
+	Permissions string `json:"permissions"`
+
+	// Content is the actual content of the file.
+	Content string `json:"content"`
+}
```

Comment:
There are a couple other optional fields provided by cloud-init, should we also expose those?

`encoding` and `append` are the ones I'm thinking of and I'm not completely convinced that we necessarily *should* expose them.

Reply:
How about we leave them out for now, and we can add them later if needed?